barrier.go

```diff
@@ -89,7 +89,7 @@ type idleBarrier struct {
 	timer *time.Timer
 	wg    sync.WaitGroup
 	outs  []edge.StatsEdge
-	stopC chan bool
+	stopC chan interface{}
```

Comment:
This should be `chan struct{}` as `struct{}` will be an empty value and the compiler will optimize it out.  On the other hand an interface needs room for a pointer to type information and a pointer to the instance.